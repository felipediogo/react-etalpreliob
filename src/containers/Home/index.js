import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import Container from '../../components/Container';

export class Home extends PureComponent {
  handleClick = async () => {
    await this.props.dispatch.count.incrementAsync();
  }

  render() {
    const { value } = this.props.count;

    return (
      <Container>
        Hello :D <br />
        <button onClick={this.handleClick}>+</button> {value}
      </Container>
    );
  }
}

const mapState = state => ({ count: state.count });

const mapDispatch = dispatch => ({ dispatch });

export default connect(mapState, mapDispatch)(Home);
