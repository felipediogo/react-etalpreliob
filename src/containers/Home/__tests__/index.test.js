import React from "react";
import { render, fireEvent } from "react-testing-library";

import Home from "../";

describe("Home", () => {
  test("default state", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  test("increment counter", () => {
    const { container, getByTestId } = render(<Home />);

    expect(getByTestId("counter")).toHaveTextContent("0");
    fireEvent.click(getByTestId("increment-button"));
    expect(getByTestId("counter")).toHaveTextContent("1");
    fireEvent.click(getByTestId("increment-button"));
    expect(getByTestId("counter")).toHaveTextContent("2");

    expect(container).toMatchSnapshot();
  });

  test("decrement counter", () => {
    const { container, getByTestId } = render(<Home />);

    expect(getByTestId("counter")).toHaveTextContent("0");
    fireEvent.click(getByTestId("decrement-button"));
    expect(getByTestId("counter")).toHaveTextContent("-1");
    fireEvent.click(getByTestId("decrement-button"));
    expect(getByTestId("counter")).toHaveTextContent("-2");

    expect(container).toMatchSnapshot();
  });
});
