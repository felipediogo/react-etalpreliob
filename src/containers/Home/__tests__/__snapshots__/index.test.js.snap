// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home decrement counter 1`] = `
<div>
  <div
    class="css-jt3klv"
  >
    Hello :D 
    <br />
    <button
      data-testid="increment-button"
    >
      +
    </button>
    <span
      data-testid="counter"
    >
      -2
    </span>
    <button
      data-testid="decrement-button"
    >
      -
    </button>
  </div>
</div>
`;

exports[`Home default state 1`] = `
<div>
  <div
    class="css-jt3klv"
  >
    Hello :D 
    <br />
    <button
      data-testid="increment-button"
    >
      +
    </button>
    <span
      data-testid="counter"
    >
      0
    </span>
    <button
      data-testid="decrement-button"
    >
      -
    </button>
  </div>
</div>
`;

exports[`Home increment counter 1`] = `
<div>
  <div
    class="css-jt3klv"
  >
    Hello :D 
    <br />
    <button
      data-testid="increment-button"
    >
      +
    </button>
    <span
      data-testid="counter"
    >
      2
    </span>
    <button
      data-testid="decrement-button"
    >
      -
    </button>
  </div>
</div>
`;
